### ISSUE TITLE
Repeated error message

### ISSUE THREAD

**Issue by wsnoble** [percolator/percolator#336]
I ran Percolator on the attached pin file, and it gave a warning message ("Warning: Set decoy prefix don't match") over and over again.  I think this is because the file uses DECOY_ instead of decoy_ to indicate decoys.  Is that what's going on?  Right now, I can't tell what Percolator is doing.  I think ithe error message should be made more clear and should only be issued once. Do I need to change the pin file to make it work correctly?

[XLpeplib_Beveridge_QEx-HFX_DSS_R1.perc.intra.txt](https://github.com/percolator/percolator/files/9170147/XLpeplib_Beveridge_QEx-HFX_DSS_R1.perc.intra.txt)

**Comment by wsnoble** [comment]
I just noticed that the first line of the stderr output is

```
Protein decoy-preix used is Cas9
``

If we ignoring the typo ("preix"), this message seems to indicate that Percolator is misidentifying the decoy precursor on the protein IDs.  I find this very strange, because the first line is a decoy with ID "DECOY__Cas9."  Why would it think "Cas9" is the prefix?

**Comment by wsnoble** [comment]
FYI, I later realized that I could specify the decoy prefix with `--protein-decoy-pattern DECOY_`, but this does not eliminate the problematic "Protein decoy-preix" message nor the repeated message about "set decoy prefix don't match."

**Comment by percolator** [comment]
I made a pull request aiming to remove these problems:
https://github.com/percolator/percolator/pull/338


(base) lukask@shannon:/tmp/xxx/build/percolator-noxml/src$ ./percolator ~/Downloads/XLpeplib_Beveridge_QEx-HFX_DSS_R1.perc.intra.txt > /dev/null
Protein decoy-preix used is Cas9
All files have been read
Percolator version 3.06.0, Build Date Aug 11 2022 17:32:55
Copyright (c) 2006-9 University of Washington. All rights reserved.
Written by Lukas Käll (lukall@u.washington.edu) in the
Department of Genome Sciences at the University of Washington.
Issued command:
./percolator /home/lukask/Downloads/XLpeplib_Beveridge_QEx-HFX_DSS_R1.perc.intra.txt
Started Thu Aug 11 17:51:14 2022
Hyperparameters: selectionFdr=0.01, Cpos=0, Cneg=0, maxNiter=10
Reading tab-delimited input from datafile /home/lukask/Downloads/XLpeplib_Beveridge_QEx-HFX_DSS_R1.perc.intra.txt
Features:
Score dScore negLog10eVal negLog10eValA negLog10eValB IonMatch ConIonMatch IonMatchA ConIonMatchA IonMatchB ConIonMatchB NormRank PPScoreDiff Charge1 Charge2 Charge3 Charge4 Charge5 Charge6 Charge7 Charge8Plus Mass PPM LenShort LenLong LenSum 
Warning: Set decoy prefix don't match
Found 1881 PSMs
Separate target and decoy search inputs detected, using mix-max method.
Train/test set contains 1473 positives and 408 negatives, size ratio=3.61029 and pi0=1
Warning: The mix-max procedure is not well behaved when # targets (1473) != # decoys (408). Consider using target-decoy competition (-Y flag).
Selecting Cpos by cross-validation.
Selecting Cneg by cross-validation.
Split 1:	Selected feature 1 as initial direction. Could separate 648 training set positives with q<0.01 in that direction.
Split 2:	Selected feature 1 as initial direction. Could separate 650 training set positives with q<0.01 in that direction.
Split 3:	Selected feature 1 as initial direction. Could separate 638 training set positives with q<0.01 in that direction.
Found 906 test set positives with q<0.01 in initial direction
Reading in data and feature calculation took 0.0445 cpu seconds or 1 seconds wall clock time.
---Training with Cpos selected by cross validation, Cneg selected by cross validation, initial_fdr=0.01, fdr=0.01
Iteration 1:	Estimated 1035 PSMs with q<0.01
Iteration 2:	Estimated 1044 PSMs with q<0.01
Iteration 3:	Estimated 1043 PSMs with q<0.01
Iteration 4:	Estimated 1043 PSMs with q<0.01
Iteration 5:	Estimated 1043 PSMs with q<0.01
Iteration 6:	Estimated 1043 PSMs with q<0.01
Iteration 7:	Estimated 1041 PSMs with q<0.01
Iteration 8:	Estimated 1041 PSMs with q<0.01
Iteration 9:	Estimated 1041 PSMs with q<0.01
Iteration 10:	Estimated 1041 PSMs with q<0.01
Learned normalized SVM weights for the 3 cross-validation splits:
 Split1	 Split2	 Split3	FeatureName
-0.7548	-0.4339	 0.0909	Score
 0.3697	 0.3731	 0.3724	dScore
 1.3441	 2.9897	 1.8469	negLog10eVal
 1.1601	 0.9385	 1.6725	negLog10eValA
 0.1022	 0.5347	 0.1262	negLog10eValB
 0.7595	 0.5896	 0.6419	IonMatch
-1.3070	-2.0979	-0.3530	ConIonMatch
 1.2202	 0.8086	 0.8724	IonMatchA
 1.4227	 2.1809	 1.5444	ConIonMatchA
-0.1239	 0.1159	 0.1382	IonMatchB
 0.9094	 1.3437	 0.9867	ConIonMatchB
-0.5932	-0.4905	-0.2419	NormRank
 0.1799	-1.2292	-0.6387	PPScoreDiff
 0.0000	 0.0000	 0.0000	Charge1
-0.0630	-0.0519	-0.0912	Charge2
-0.0967	-0.0725	 0.1281	Charge3
 0.1014	 0.0807	-0.0920	Charge4
 0.0796	 0.0459	-0.0279	Charge5
-0.1262	-0.1040	-0.1826	Charge6
 0.0000	 0.0000	 0.0000	Charge7
 0.0000	 0.0000	 0.0000	Charge8Plus
-1.0982	-1.1803	-1.4298	Mass
 0.1789	 0.1244	 0.1747	PPM
 0.2781	-0.0453	 0.0933	LenShort
 0.4578	 0.4143	 0.5640	LenLong
 0.4594	 0.2688	 0.4426	LenSum
 1.9317	 1.5915	 2.7946	m0
Found 989 test set PSMs with q<0.01.
Tossing out "redundant" PSMs keeping only the best scoring PSM for each unique peptide.
Selecting pi_0=0.428849
Calculating q values.
New pi_0 estimate on final list yields 356 target peptides with q<0.01.
Calculating posterior error probabilities (PEPs).
Processing took 2.4520 cpu seconds or 1 seconds wall clock time.
(base) lukask@shannon:/tmp/xxx/build/percolator-noxml/src$
